homeassistant/components/lyric/sensor.py

```diff
@@ -0,0 +1,254 @@
+"""Support for Honeywell Lyric sensor platform."""
+from datetime import datetime, timedelta
+
+from aiolyric.objects.device import LyricDevice
+from aiolyric.objects.location import LyricLocation
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    DEVICE_CLASS_HUMIDITY,
+    DEVICE_CLASS_TEMPERATURE,
+    DEVICE_CLASS_TIMESTAMP,
+)
+from homeassistant.helpers.typing import HomeAssistantType
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+from homeassistant.util import dt as dt_util
+
+from . import LyricDeviceEntity
+from .const import (
+    DOMAIN,
+    PRESET_HOLD_UNTIL,
+    PRESET_NO_HOLD,
+    PRESET_PERMANENT_HOLD,
+    PRESET_TEMPORARY_HOLD,
+    PRESET_VACATION_HOLD,
+)
+
+SCAN_INTERVAL = timedelta(seconds=120)
+PARALLEL_UPDATES = 6
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up the Honeywell Lyric sensor platform based on a config entry."""
+    coordinator: DataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    entities = []
+
+    for location in coordinator.data.locations:
+        for device in location.devices:
+            if device.indoorTemperature:
+                entities.append(
+                    LyricIndoorTemperatureSensor(hass, coordinator, location, device)
+                )
+            if device.outdoorTemperature:
+                entities.append(
+                    LyricOutdoorTemperatureSensor(hass, coordinator, location, device)
+                )
+            if device.displayedOutdoorHumidity:
+                entities.append(
+                    LyricOutdoorHumiditySensor(hass, coordinator, location, device)
+                )
+            if device.changeableValues:
+                if device.changeableValues.nextPeriodTime:
+                    entities.append(
+                        LyricNextPeriodSensor(hass, coordinator, location, device)
+                    )
+                if device.changeableValues.thermostatSetpointStatus:
+                    entities.append(
+                        LyricSetpointStatusSensor(hass, coordinator, location, device)
+                    )
```

Comment:
set `cls` and then do

if cls is not None:
   entities.append(
     cls(hass, coordinator, location, device)
   )

Reply:
Not sure this is necessary, as I'll still be appending entities anyway?

```py
    for location in coordinator.data.locations:
        for device in location.devices:
            if device.indoorTemperature:
                cls = LyricIndoorTemperatureSensor
                entities.append(cls(hass, coordinator, location, device))
            if device.outdoorTemperature:
                cls = LyricOutdoorTemperatureSensor
                entities.append(cls(hass, coordinator, location, device))
            if device.displayedOutdoorHumidity:
                cls = LyricOutdoorHumiditySensor
                entities.append(cls(hass, coordinator, location, device))
            if device.changeableValues:
                if device.changeableValues.nextPeriodTime:
                    cls = LyricNextPeriodSensor
                    entities.append(cls(hass, coordinator, location, device))
                if device.changeableValues.thermostatSetpointStatus:
                    cls = LyricSetpointStatusSensor
                    entities.append(cls(hass, coordinator, location, device))
```

If I add this to the end, only the last class gets counted. Unless there's a better way I'm missing here? :confused:

Reply:
```py
    for location in coordinator.data.locations:
        for device in location.devices:
            cls = None
            if device.indoorTemperature:
                cls = LyricIndoorTemperatureSensor
            if device.outdoorTemperature:
                cls = LyricOutdoorTemperatureSensor
            if device.displayedOutdoorHumidity:
                cls = LyricOutdoorHumiditySensor
            if device.changeableValues:
                if device.changeableValues.nextPeriodTime:
                    cls = LyricNextPeriodSensor
                if device.changeableValues.thermostatSetpointStatus:
                    cls = LyricSetpointStatusSensor
            if cls:
                entities.append(cls(hass, coordinator, location, device))
```

If I add this to the end, only the last class gets counted. Unless there's a better/smarter way I'm missing here? :confused: